src/nupic/algorithms/SpatialPooler.hpp

```diff
@@ -1248,8 +1245,8 @@ class SpatialPooler : public Serializable
   bool wrapAround_;
   UInt updatePeriod_;
 
-  Real synPermMin_;
-  Real synPermMax_;
+  Real synPermMin_ = 0.0f;
```

Comment:
and same here, 
does MSVC really complain about `Real a = 0;`? It should be able to infer the exact type at compile-time. That's what all other compilers do. 

And if it complains, I think we should do `Real a = (Real)0;` instead (of `= 0.0f`) because the (T)cast is more flexible if you decided to change `using Real = double` you wouldn;t have to change all literals in codebase again.

Reply:
MSVC does handle 0 ok. But 5.0 is assumed to be a double and if being assigned to a float it is a type conversion.

My personal feeling is that all types should be specific and locked in.  CSharp requires it.  Python goes the other way and allows anything because values carry type at runtime.  But Python pays for that flexibility in execution time.